BraveWallet/Settings/CustomNetworkDetailsView.swift

```diff
@@ -0,0 +1,458 @@
+// Copyright 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import SwiftUI
+import BraveCore
+import Shared
+
+struct NetworkInputItem: Identifiable {
+  var input: String
+  var error: String?
+  var id = UUID()
+}
+
+struct NetworkTextField: View {
+  var placeholder: String
+  @Binding var item: NetworkInputItem
+  
+  var body: some View {
+    VStack(alignment: .leading) {
+      TextField(placeholder, text: $item.input)
+        .autocapitalization(.none)
+        .disableAutocorrection(true)
+      if let error = item.error {
+        HStack(alignment: .firstTextBaseline, spacing: 4) {
+          Image(systemName: "exclamationmark.circle.fill")
+          Text(error)
+            .fixedSize(horizontal: false, vertical: true)
+              .animation(nil, value: error)
+        }
+        .accessibilityElement(children: .combine)
+        .transition(
+          .asymmetric(
+            insertion: .opacity.animation(.default),
+            removal: .identity
+          )
+        )
+        .font(.footnote)
+        .foregroundColor(Color(.braveErrorLabel))
+      }
+    }
+  }
+}
+
+class CustomNetworkModel: ObservableObject {
+  @Published var networkId = NetworkInputItem(input: "") {
+    didSet {
+      if networkId.input != oldValue.input {
```

Comment:
Is there really any reason for all of these equality checks on all of the values here?

Reply:
in didSet we validate the input and set error message if the input is invalid. 
However, set error message will also trigger didSet since `NetworkInputItem` is a struct. This will cause a infinite loop.

Reply:
😱 Ah I see. Okay for now I think we'll have to keep those 😂